Temporary files can have fields at their creation (#5535)
qgis/QGIS-Documentation: docs/user_manual/managing_data_source/create_layers.rst

   * ``Point`` or ``MultiPoint`` layer,
   * ``LineString/CompoundCurve`` or ``MultiLineString/MultiCurve`` layer,
   * ``Polygon/CurvePolygon`` or ``MultiPolygon/MultiSurface`` layer.

> #. If of a geometric type, specify the dimensions of the dataset: check whether

Reviewer: ```suggestion
#. For geometric types, specify the dimensions of the dataset: check whether
```

   it should :guilabel:`Include Z dimension` and/or :guilabel:`Include M values`
#. Specify the coordinate reference system using the |setProjection| button.
#. Add fields to the layer. Note that unlike many formats, temporary
   layers can be created without any fields. This step is thus optional.

   #. Enter the :guilabel:`Name` of the field
   #. Select the data :guilabel:`Type`: :guilabel:`Text`, :guilabel:`Whole number`,
      :guilabel:`Decimal number`, :guilabel:`Boolean`, :guilabel:`Date`,
      :guilabel:`Time` and :guilabel:`Date & Time` are supported.
   #. Depending on the selected data format, enter the :guilabel:`Length` and
      :guilabel:`Precision`
   #. Click on the |newAttribute| :guilabel:`Add to Fields List` button

>    #. Reproduce the steps above for each field you need to add

Reviewer: ```suggestion
   #. Repeat the steps above for each field you need to add
```
